frontend/src/app/shared/components/file-links/file-picker-modal/sort-files.pipe.ts

```diff
@@ -0,0 +1,25 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { isDirectory } from 'core-app/shared/components/file-links/file-link-icons/file-icons.helper';
+
+@Pipe({
+  name: 'sortFiles',
+})
+export class SortFilesPipe implements PipeTransform {
+  transform<T extends { mimeType?:string, name:string }>(array:T[]):T[] {
+    return array.sort((a, b):number => {
+      if (isDirectory(a.mimeType) && isDirectory(b.mimeType)) {
+        return a.name.localeCompare(b.name);
+      }
+
+      if (isDirectory(a.mimeType)) {
+        return -1;
+      }
+
+      if (isDirectory(b.mimeType)) {
+        return 1;
+      }
+
+      return a.name.localeCompare(b.name);
```

Comment:
Nitpicking but isn't this just a weird way of writing an `if ... else if ...`

Reply:
The reason I did it like that is, that usually comparer functions tend to be hard to understand. I wanted to make clear, what cases exist and have that clearly visible without noise. I explicitly chose this style because of less code noise. One can clearly follow what gets evaluated one after another. That was my goal.

IMHO the implementation would not gain much readability with `else if`-chaining or introducing a "result" variable that is returned at the end.

Reply:
I'm a big fan of the early return pattern as well, I'd keep it this way.